### PR TITLE
Update to API v2

### PR DESCRIPTION
**Pull request by vsiul**
- Library updated to support M2X API version 2
- Library file renamed to use a common versioning method. (m2x-MAJOR.MINOR.PATCH.js).
- M2XUVDemo updated to use API v2. M2XGPSDemo app is still using the previous version of the library (version 1) for illustration purposes.
- Documentation updated
- Cosmetic and typos fixed

### CODE REVIEW

File: m2x-2.0.0.js [hunk 1]
```diff
@@ -0,0 +1,1225 @@
...
+             * @param {Object} [options] Search options. See https://m2x.att.com/developer/documentation/v2/device#List-Search-Devices for details.
+             * @param {Function} onSuccess Function to call back if operation is successful
+             * @param {Function} [onError] Function to call back if operation fails
+             */
+            search : function (options,cbS,cbE) {
+                var onSuccess = cbS,
+                    onError = cbE;
+                
+                if (Utils.isFunc(options)) {
+                    onSuccess = options;
```
Comment (grilix) [inline m2x-2.0.0.js]: We also need to empty `options` here, since it's a function.

File: m2x-2.0.0.js [hunk 2]
```diff
@@ -0,0 +1,1225 @@
...
+             * @param {Object} [options] Search options. See https://m2x.att.com/developer/documentation/v2/device#List-Device-Groups for details.
+             * @param {Function} onSuccess Function to call back if operation is successful
+             * @param {Function} [onError] Function to call back if operation fails
+             */
+            searchDeviceGroups : function (options,cbS,cbE) {
+                var onSuccess = cbS,
+                    onError = cbE;
+                
+                if (Utils.isFunc(options)) {
+                    onSuccess = options;
```
Comment (grilix) [inline m2x-2.0.0.js]: Empty `options` here, too.

**Comment by xxuejie** [comment]
It appears to us that you have updated the UV demo to V2, but not the GPS demo. Any particular reason for this? Any chance you want to update the GPS demo as well?

**Comment by vsiul** [comment]
GPS demo is using v1 API only for illustration purposes (i.e. some developer might want to try API v1 and this could help as a starting point)

File: m2x-2.0.0.js (hunk 1 above)
Comment (vsiul) [inline m2x-2.0.0.js]: ```options``` is an optional argument. You can see across the code that several APIs receive an options argument. If the argument is a function (checked by ```Utils.isFunc(options)``` ) then we know the user didn't provide search options.

Comment (grilix) [inline m2x-2.0.0.js]: You are right, but if `options` is a function, you will end up passing that function in `urlparams` below, right?

Comment (vsiul) [inline m2x-2.0.0.js]: Yeah, you're right! I'm fixing the code right now. I completely missed that. Good catch!

Comment (Adonya) [inline m2x-2.0.0.js]: Thanks Josh!

On Fri, Jan 2, 2015 at 9:58 AM, vsiul <notifications@github.com> wrote:

> In m2x-2.0.0.js
> <https://github.com/attm2x/m2x-gear-s/pull/1#discussion-diff-22416812>:
>
> > +         */
> > +        api.device = {
> > +            /**
> > +             * Retrieve the list of devices accessible by the authenticated API key that meet the search criteria.
> > +             * @method search
> > +             * @async
> > +             * @param {Object} [options] Search options. See https://m2x.att.com/developer/documentation/v2/device#List-Search-Devices for details.
> > +             * @param {Function} onSuccess Function to call back if operation is successful
> > +             * @param {Function} [onError] Function to call back if operation fails
> > +             */
> > +            search : function (options,cbS,cbE) {
> > +                var onSuccess = cbS,
> > +                    onError = cbE;
> > +
> > +                if (Utils.isFunc(options)) {
> > +                    onSuccess = options;
>
> Yeah, you're right! I'm fixing the code right now. I completely missed
> that. Good catch!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/attm2x/m2x-gear-s/pull/1/files#r22416812>.
>



-- 
Adonya Ourshalimian
(541) 292-2422

**Comment by grilix** [comment]
Awesome! Looks good to me :+1:

**Pull request merged**